Mathlib/Analysis/Calculus/FDeriv/Symmetric.lean

```diff
@@ -441,37 +465,73 @@ theorem second_derivative_symmetric_of_eventually {f' : E → E →L[𝕜] F} {x
 
 /-- If a function is differentiable, and has two derivatives at `x`, then the second
 derivative is symmetric. -/
-theorem second_derivative_symmetric {f' : E → E →L[𝕜] F} {f'' : E →L[𝕜] E →L[𝕜] F} {x : E}
+theorem second_derivative_symmetric [IsRCLikeNormedField 𝕜]
+    {f' : E → E →L[𝕜] F} {f'' : E →L[𝕜] E →L[𝕜] F} {x : E}
     (hf : ∀ y, HasFDerivAt f (f' y) y) (hx : HasFDerivAt f' f'' x) (v w : E) : f'' v w = f'' w v :=
   second_derivative_symmetric_of_eventually (Filter.Eventually.of_forall hf) hx v w
 
-/-- If a function is `C^2` at a point, then its second derivative there is symmetric. -/
-theorem ContDiffAt.isSymmSndFDerivAt {n : WithTop ℕ∞} (hf : ContDiffAt 𝕜 n f x) (hn : 2 ≤ n) :
-    IsSymmSndFDerivAt 𝕜 f x := by
-  intro v w
-  apply second_derivative_symmetric_of_eventually (f := f) (f' := fderiv 𝕜 f) (x := x)
-  · obtain ⟨u, hu, h'u⟩ : ∃ u ∈ 𝓝 x, ContDiffOn 𝕜 2 f u :=
-      (hf.of_le hn).contDiffOn (m := 2) le_rfl (by simp)
-    rcases mem_nhds_iff.1 hu with ⟨v, vu, v_open, xv⟩
-    filter_upwards [v_open.mem_nhds xv] with y hy
-    have : DifferentiableAt 𝕜 f y := by
-      have := (h'u.mono vu y hy).contDiffAt (v_open.mem_nhds hy)
-      exact this.differentiableAt one_le_two
-    exact DifferentiableAt.hasFDerivAt this
-  · have : DifferentiableAt 𝕜 (fderiv 𝕜 f) x := by
-      apply ContDiffAt.differentiableAt _ le_rfl
-      exact hf.fderiv_right hn
-    exact DifferentiableAt.hasFDerivAt this
+open scoped Classical in
+variable (𝕜) in
+/-- `minSmoothness 𝕜 n` is the minimal smoothness exponent larger than `n` for which one can do
+serious calculus in `𝕜`. If `𝕜` is `ℝ` or `ℂ`, this is just `n`. Otherwise, this is `ω` as only
+analytic functions are well behaved on `ℚₚ`, say. -/
+noncomputable irreducible_def minSmoothness (n : WithTop ℕ∞) :=
+  if IsRCLikeNormedField 𝕜 then n else ω
+
+@[simp] lemma minSmoothness_of_IsRCLikeNormedField [h : IsRCLikeNormedField 𝕜] {n : WithTop ℕ∞} :
+    minSmoothness 𝕜 n = n := by
+  simp [minSmoothness, h]
+
+lemma le_minSmoothness {n : WithTop ℕ∞} : n ≤ minSmoothness 𝕜 n := by
+  simp only [minSmoothness]
+  split_ifs <;> simp
+
+lemma exist_minSmoothness_le_ne_infty {n : WithTop ℕ∞} {m : ℕ} (hm : minSmoothness 𝕜 m ≤ n) :
```

Comment:
It's getting late, and my brain stared at this for too long. To ensure I got it: this lemma makes sense over general fields, because the \infty stands for "smooth, but not analytic", right?

Perhaps a comment is useful. Up to you.